moped-editor/src/views/projects/projectView/ProjectWorkActivity/ProjectWorkActivityTable.js

```diff
@@ -172,16 +179,35 @@ const ProjectWorkActivitiesTable = () => {
 
   const onClickAddActivity = () => setEditActivity({ project_id: projectId });
 
+  // const onDeleteActivity = useCallback(
+  //   ({ id }) => {
+  //     window.confirm("How is it going?") &&
+  //       deleteContract({ variables: { id } }).then(() => {
+  //         refetch();
+  //       });
+  //   },
+  //   [deleteContract, refetch]
+  // );
+
+  // new code /////
   const onDeleteActivity = useCallback(
     ({ id }) => {
-      window.confirm("Are you sure you want to delete this activity?") &&
-        deleteContract({ variables: { id } }).then(() => {
-          refetch();
-        });
+      setActivityToDelete(id);
+      setIsDeleteConfirmationOpen(true);
     },
-    [deleteContract, refetch]
+    []
   );
 
+  const handleDeleteConfirmed = useCallback(() => {
```

Comment:
Just want to comment that the React tutorials served you well! `handleDeleteConfirmed` is an improved name for this function that [responds to an event](https://react.dev/learn/responding-to-events). 🚀